File: Groove.CSFE/Groove.CSFE.Core/Entities/CarrierModel.cs
namespace Groove.CSFE.Core.Entities
{
    public class CarrierModel : Entity
    {
        public long Id { get; set; }
        public string CarrierCode { get; set; }
        public int? CarrierNumber { get; set; }
        public string ModeOfTransport { get; set; }
        public string Name { get; set; }
        public CarrierStatus Status { get; set; }
    }
}

File: Groove.SP/groove-sp-web/src/app/core/models/carrier.model.ts
import { CarrierStatus } from "./enums/enums";

export class CarrierModel {
    id: number;
    carrierCode: string;
    carrierNumber: number;
    modeOfTransport: string;
    name: string;
    status: CarrierStatus;
}

export class CarrierSelectionModel extends CarrierModel {
    displayName: string;
    /**
     *
     */
    constructor(carrier: CarrierModel) {
        super();
        this.modeOfTransport = carrier.modeOfTransport
        this.name = carrier.name;
        this.carrierCode = carrier.carrierCode;
        this.displayName = `${carrier.carrierCode} - ${carrier.name}`;
    }
}